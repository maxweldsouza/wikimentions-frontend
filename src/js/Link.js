import React from 'react';

const Link = ({ type, tab, id, slug, className, children }) => {
    let pagepath;
    let defaultTab;
    if (type === 'video') {
        pagepath = '/videos/';
        defaultTab = 'mentioned';
    } else if (type === 'book') {
        pagepath = '/books/';
        defaultTab = 'mentioned';
    } else if (type === 'person') {
        pagepath = '/people/';
        defaultTab = 'videos';
    } else {
        throw new Error('No page type specified');
    }

    const finaltab = tab ? tab : defaultTab;
    let href;
    let nofollow = false;
    if (['edit', 'discuss', 'history'].includes(finaltab)) {
        href = `/${finaltab}/${id}/${slug}`;
        nofollow = true;
    } else if (finaltab !== defaultTab) {
        href = `${pagepath + id}/${slug}/${finaltab}`;
    } else {
        href = `${pagepath + id}/${slug}`;
    }
    if (nofollow) {
        return (
            <a rel="nofollow" className={className} href={href}>
                {children}
            </a>
        );
    }
    return <a className={className} href={href}>{children}</a>;
};

export default Link;
