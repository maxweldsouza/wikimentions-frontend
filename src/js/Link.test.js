import React from 'react';
import renderer from 'react-test-renderer';
import Link from './Link';

test('Link renders correctly', () => {
    ['person', 'video', 'book'].map(type => {
        const tree = renderer
            .create(<Link type={type} id={1} slug="target-name" />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
