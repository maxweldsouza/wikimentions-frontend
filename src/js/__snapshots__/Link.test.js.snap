// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders correctly 1`] = `
<a
  className={undefined}
  href="/people/1/target-name"
/>
`;

exports[`Link renders correctly 2`] = `
<a
  className={undefined}
  href="/videos/1/target-name"
/>
`;

exports[`Link renders correctly 3`] = `
<a
  className={undefined}
  href="/books/1/target-name"
/>
`;
